src/adic_space.lean

```diff
@@ -8,11 +8,132 @@ import for_mathlib.topology
 
 open nat function
 
-variables (R : Type) [comm_ring R] [topological_space R] [topological_ring R]  
+section comm_ring
+variables (R : Type) [comm_ring R]
+-- This section is filled in in Johan's PR
+definition is_subring {R : Type} [comm_ring R] : set R → Prop := sorry 
+definition is_integrally_closed {R : Type} [comm_ring R] : set R → Prop := sorry  
+end comm_ring
+
+section topological_ring
+
+variables (R : Type*) [comm_ring R] [topological_space R] [comm_ring R] [topological_ring R]  
+
+def nhd_zero := (nhds (0 : R)).sets
+variable {R}
+lemma nhd_zero_symmetric {V : set R} : V ∈ nhd_zero R →  (λ a, -a) '' V ∈ nhd_zero R :=
+begin
+  intro H,
+  have := continuous.tendsto (topological_add_group.continuous_neg R) 0,
+  rw (show (λ (a : R), -a) 0 = 0, by simp) at this,
+  have almost:= this H,
+  have aux : { r : R | -r ∈ V } = (λ a, -a) '' V, by simp[set.image, neg_eq_iff_neg_eq],
+  simpa [filter.mem_map, aux] using almost
+end
+
+-- peredicates we need for topological rings
+
+-- Following is copy-pasted from t2_space class.
+-- We need to think whether we could directly use that class
+definition is_hausdorff (α : Type*) [topological_space α] : Prop := 
+  ∀x y, x ≠ y → ∃u v : set α, is_open u ∧ is_open v ∧ x ∈ u ∧ y ∈ v ∧ u ∩ v = ∅
```

Comment:
I would prefer uppercase for the open subsets. But of course that is only a matter of notation. Also, does it make sense to use `opens X` in this case?